college_flow_app/lib/shared/widgets/header.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:college_flow_app/shared/widgets/gap.dart';
+import 'package:flutter/material.dart';
+
+import '../../config/design_system/data/colors/colors.dart';
+import '../../config/design_system/data/spacing/spacing.dart';
+import 'flow_icon.dart';
+import 'flow_icon_button.dart';
+
+class Header extends StatelessWidget {
+  const Header({
+    Key? key,
+    this.description,
+    this.title = "",
+    this.textAlign = TextAlign.start,
+  }) : super(
+          key: key,
+        );
+  final String? description;
+  final String title;
+  final TextAlign textAlign;
+  bool get hasDescription => description != null;
```

Comment:
you can also try to remove the getter and set the default value to description to "", and set this value to text. Than if is empty the textField won't occupate any space, and you will make your logic simpler.